internal/state/store.go

```diff
@@ -296,17 +326,19 @@ func (store dbStore) Bootstrap(state State) error {
 // This will cause some old states to be left behind when doing incremental partial prunes,
 // specifically older checkpoints and LastHeightChanged targets.
 func (store dbStore) PruneStates(from int64, to int64, evidenceThresholdHeight int64) error {
+	defer addTimeSample(store.metrics.StoreAccessDurationSeconds.With("method", "prune_states"), time.Now())()
 	if from <= 0 || to <= 0 {
 		return fmt.Errorf("from height %v and to height %v must be greater than 0", from, to)
 	}
 	if from >= to {
 		return fmt.Errorf("from height %v must be lower than to height %v", from, to)
 	}
 
-	valInfo, err := loadValidatorsInfo(store.db, min(to, evidenceThresholdHeight))
+	valInfo, elapsedTime, err := loadValidatorsInfo(store.db, min(to, evidenceThresholdHeight))
```

Comment:
I'm not sure about the patterns used in our codebase, but it looks to me that `loadValidatorsInfo` doesn't benefit from `elapsedTime` at all and maybe the measurement should happen in this (`PruneStates`) function, not in the `loadValidatorsInfo`.

I can see that the unmarshaling of the data from the database is not measured, which gives us a more precise measurement for database activity. However, this is not indicated in `loadValidatorsInfo`. Maybe a solution is to document that the result is the elapsed time for _database access_ or something. But maybe we should move this measurement to `PruneStates` and accept that the unmarshaling is also part of the measurement.

Reply:
The point was to avoid measuring marshalling/unmarshalling to make sure the measurements include only DB access time. You are however right that the function `loadValidatorsInfo` itself will not report this. The function has no object that has metrics associated with it to enter the value. That is why I report the time spent in this function only within functions that call it.